src/main/java/com/redhat/exhort/utils/PythonControllerBase.java

```diff
@@ -304,96 +277,135 @@ private List<Map<String, Object>> getDependenciesImpl(
-    return getPipCommandInvokedOrDecodedFromEnvironment(
-        "EXHORT_PIP_SHOW", pipBinaryLocation, "show", depNames);
-    //    return Operations.runProcessGetOutput(pythonEnvironmentDir, pipBinaryLocation, "show",
-    // depNames);
+    return executeCommandOrExtractFromEnv("EXHORT_PIP_SHOW", pipBinaryLocation, "show", depNames);
   }
 
   String getPipFreezeFromEnvironment() {
-    return getPipCommandInvokedOrDecodedFromEnvironment(
+    return executeCommandOrExtractFromEnv(
         "EXHORT_PIP_FREEZE", pipBinaryLocation, "freeze", "--all");
   }
 
-  private String getPipCommandInvokedOrDecodedFromEnvironment(String EnvVar, String... cmdList) {
-    return getStringValueEnvironment(EnvVar, "").trim().equals("")
-        ? Operations.runProcessGetOutput(pythonEnvironmentDir, cmdList)
-        : new String(Base64.getDecoder().decode(getStringValueEnvironment(EnvVar, "")));
+  List<PythonDependency> getDependencyTreeJsonFromPipDepTree() {
+    executeCommandOrExtractFromEnv(
+        "EXHORT_PIP_PIPDEPTREE", pipBinaryLocation, "install", "pipdeptree");
+
+    String pipdeptreeJsonString = "";
+    if (isVirtualEnv()) {
+      pipdeptreeJsonString =
+          executeCommandOrExtractFromEnv("EXHORT_PIP_PIPDEPTREE", "./bin/pipdeptree", "--json");
+    } else if (isRealEnv()) {
+      pipdeptreeJsonString =
+          executeCommandOrExtractFromEnv(
+              "EXHORT_PIP_PIPDEPTREE", pathToPythonBin, "-m", "pipdeptree", "--json");
+    }
+    if (debugLoggingIsNeeded()) {
+      String pipdeptreeMessage =
+          String.format(
+              "Package Manager pipdeptree --json command result output -> %s %s",
+              System.lineSeparator(), pipdeptreeJsonString);
+      log.info(pipdeptreeMessage);
+    }
+    return mapToPythonDependencies(pipdeptreeJsonString);
+  }
+
+  public static List<PythonDependency> mapToPythonDependencies(String jsonString) {
+    try {
+      // Parse JSON and store in a list of JsonNodes
+      List<JsonNode> jsonNodeList = new ArrayList<>();
+      new ObjectMapper().readTree(jsonString).elements().forEachRemaining(jsonNodeList::add);
+
+      return jsonNodeList.stream()
+          .filter(JsonNode::isObject)
+          .map(
+              dependencyNode -> {
+                String name = dependencyNode.get("package").get("package_name").asText();
+                String version = dependencyNode.get("package").get("installed_version").asText();
+
+                // Extract dependencies
+                List<String> depList = new ArrayList<>();
+                dependencyNode
+                    .get("dependencies")
+                    .elements()
+                    .forEachRemaining(e -> depList.add(e.get("package_name").asText()));
+
+                return new PythonDependency(name, version, depList);
+              })
+          .collect(Collectors.toList());
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException(e);
+    }
```

Comment:
@JudeNiroshan In addition to the underlying json parsing error ( due to unexpected wrong input) , give some context to the client/user, what happend ( "for example, failed to parse pipdeptree json,............"), it's true that there is a stack trace that can point exactly to the function it happened in, but it's better to clarify that it failed on parsing pipdeptree json output, it will make it easier to troubleshoot problems of pipdeptree errors ( pipdeptree wasn't invoked correctly because couldn't find its binary on path or if it returned other error....)

Reply:
f08f60005e6dca6e1f8b2e137908775b165b460b